articles/a-beginners-guide-to-python/index.md

```diff
@@ -0,0 +1,207 @@
+```
+
+### Understanding lists
+Lists allow us to store numerous elements in a particular variable. For instance, we can have a list that stores all the student names in a class. We use `[]` to define a list.
+
+```
+students = [] #list example
+```
+
+Elements in a list are usually separated by a comma, as shown below.
+```
+students = [“john”, “Mary Thomas”, “John Smith”]
+```
+
+Each element in the above `students` list has an index. By default, the first index is 0. So the item at index [0] is `john`, while the value at index `1` is `Mary Thomas`. 
+A list of integers will look as follows.
+
+```
+student_marks =[90, 78, 90, 78]
+```
+We can access different list functionalities using built-in functions. For instance, to add a value to the `student_marks` list, we use the `append` function.
+
+```python
+student_marks.append(“Guardian Angel”)
+```
+
+The above function adds `Guardian Angel` at the end of the `student_marks` list.
+We use `len(student_marks)` to determine the length of the list.
+We use the remove() function to delete something from the list.
+
+### Understanding functions or methods
+Methods are quite critical in programming. They help store reusable code. This means that a person can call already declared methods rather than writing statements from scratch repeatedly. This saves significant time, which can be invested in other productive activities.
+
+In Python, we use the `def` keyword to declare a function. 
+An example of a python method is shown below.
+```
+def readData():
+    print('success')
+```
+The above function prints `success` when it’s invoked.
+We can also pass data to a method, perform some calculations, and return the results. This is demonstrated in the code snippet below.
+
+```
+def calculateTotal(chem, bio):
+    return chem+bio
+
+print(calculateTotal (90,80)
+```
+The `calculateTotal` method takes in two parameters (chem, bio). The function then returns the sum of the two values. It is important to take note of the data types when passing parameters. For instance, the `calculateTotal` method will not work when we pass in a string as a parameter. This is because the program cannot sum up an integer and a string. 
+As shown above, we can call the `calculateTotal` method directly from our print statement.
+```
+print(calculateTotal (90,80)
+```
+The `return` keyword ensures that the method returns a result after execution.
+
+Note that a function can also call another method. This is illustrated below.
+
+```python
+def readData(chem, bio):
+    return chem+bio
+
+def getTotal():
+    print(readData(90,80)) #calls the readData method
+
+getTotal()
+```
+
+### Understanding loops
+Loops are critical because they allow us to iterate through lists, check for different conditions, and continuously execute various statements. The main loops are `for` and `while`.
+
+1. For loops
+As noted, we can use a for loop to iterate through a list, as shown below.
+
+```
+Student_list=[“John Doore”,”Matu Smith”]
```

Comment:
```suggestion
student_list=[“John Doore”,”Matu Smith”]
```

This should be student_list with a lowercase `s`, since variables in Python are case sensitive.